add missing word for array length in docs storage layout (#13270)
ethereum/solidity: docs/internals/layout_in_storage.rst

``bytes`` and ``string``
------------------------

``bytes`` and ``string`` are encoded identically. In general, the encoding is similar to ``bytes1[]`` in the

>  sense that the layout in storage depends on the ``.length``:

Reviewer: But this is not true. The layout of `bytes1[]` does not depend on `.length`. The similarity is that the long form of `bytes` and `bytes1[]` both have a length slot and a separate data area.

> - For short values (shorter than 32 bytes) the array elements are stored together with the length in the same slot.

Reviewer 2: ```suggestion
- For short values (less than 32 bytes) the array elements are stored together with the length in the same slot.
```

> - For short values (shorter than 32 bytes) the array elements are stored together with the length in the same slot.

Reviewer: Maybe bullet points would work but in the current form you have two single-point lists, which is a bit weird.

If the data is at most ``31`` bytes long, the elements are stored
in the higher-order bytes (left aligned) and the lowest-order byte stores the value ``length * 2``.

> - For long values (more than 32 bytes), the length is stored in one slot (which position is defined as per the [storage layout rules](#layout-of-state-variables-in-storage)), while the array elements are stored in a separately allocated data area whose location is computed using a ``keccak256`` hash of that slot's position.

Reviewer 2: ```suggestion
- For long values (32 bytes or more), the length is stored in one slot (whose position is defined as per the [storage layout rules](#layout-of-state-variables-in-storage)), while the array elements are stored in a separately allocated data area whose location is computed using a ``keccak256`` hash of that slot's position.
```

> - For long values (more than 32 bytes), the length is stored in one slot (which position is defined as per the [storage layout rules](#layout-of-state-variables-in-storage)), while the array elements are stored in a separately allocated data area whose location is computed using a ``keccak256`` hash of that slot's position.

Reviewer: This is not a markdown file. RST has different syntax for links (please check the text around).

------------------------

``bytes`` and ``string`` are encoded identically.
In general, the encoding is similar to ``bytes1[]``, in the sense that there is a slot for the array length itself and
a data area that is computed using a ``keccak256`` hash of that slot's position.

> However, for short values (shorter than 32 bytes) the array elements are stored together with the length in the same slot.

Reviewer: I agree with @chriseth that the change does not necessarily clarify things because the slot is not meant only for length.

If we were to expand this, this would be my proposal:
```
In general, the encoding is similar to ``bytes1[]``, in the sense that it consists of one fixed slot and
a separately allocated data area whose location is computed using a ``keccak256`` hash of that slot's position.
However, the fixed slot is not used exclusively to store the length.
For short values (shorter than 32 bytes) the array elements are stored together with the length in the fixed slot.
```
